pkg/proxy/serviceport.go

```diff
@@ -78,10 +78,10 @@ type BaseServicePortInfo struct {
 	port                     int
 	protocol                 v1.Protocol
 	nodePort                 int
-	loadBalancerVIPs         []string
+	loadBalancerVIPs         []net.IP
```

Comment:
@uablrek we can parse the IPs, stringify again to get the canonical address, and keep the data type as string only?
I think we pass IPs as strings only to the majority of the external APIs with which kube-proxy interacts. 

or maybe an aliased type `IP` (which is just a plain string)  to distinguish it from string everywhere.

Reply:
Please read the discussion in https://github.com/kubernetes/kubernetes/pull/122620#issuecomment-1879730388 and onward

Reply:
oh, I missed that PR.